traits/ctraits.c

```diff
@@ -1796,49 +1729,32 @@ getattr_delegate ( trait_object      * trait,
         Py_INCREF( delegate );
     }
 
-    nname = Py2to3_NormaliseAttrName(name);
-
-    if( nname == NULL ){
+    if (!PyUnicode_Check(name)) {
         invalid_attribute_error( name );
         Py_DECREF( delegate );
         return NULL;
     }
 
-    delegate_attr_name = trait->delegate_attr_name( trait, obj, nname );
+    delegate_attr_name = trait->delegate_attr_name( trait, obj, name );
     tp = Py_TYPE(delegate);
 
     if ( tp->tp_getattro != NULL ) {
         result = (*tp->tp_getattro)( delegate, delegate_attr_name );
         goto done;
     }
 
-    if ( tp->tp_getattr != NULL ) {
```

Comment:
This is the only potential behaviour change in the PR: we no longer check `tp_getattr` in addition to `tp_getattro`. But `tp_getattr` is long-deprecated, and there's little reason to expect anything we care about to have a non-NULL `tp_getattr` slot.